pages/account/cards.tsx

```diff
@@ -157,8 +167,86 @@ function CardsDisplay() {
             const theData = appContext.data.getData();
             setCardsData(theData.get("cards"));
             cardSets.forEach(cardSet => { cardSet.cards = (cards) ? cards.filter(card => card.data.category == cardSet.cardSetName) : [] });
+
+            if (cards) {
```

Comment:
pretty sure this will alwaya be true, as you are likely initializing it as `[]`.

Maybe check that length > 0?

I've had instances where I loaded the page and saw an empty list of filters and had to refresh to fix it, not sure if it' related or not but could be.

Reply:
This wasn't initialized as [] so it was a Card[] | undefined, so I added that to remove all the warnings about it maybe being undefined and the '?' necessary for that

Added an empty array for initialization so it's cleaner

Also added some dependency for useEffect and useMemo to help get the list of bonus more consistently